build.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+# ****************************************************************************
+# Copyright 2004-2013 Castle Project - http://www.castleproject.org/
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ****************************************************************************
+
+if [ ! -d "./.dotnet" ]; then
+
+    sudo apt-get -qq update
```

Comment:
Yes, in build matrix we will have one for Ubuntu and another for OSX. It will defo tell you which failed in the lower matrix and roll that up to the main build.